skel/handlers/middlewares.go

```diff
@@ -12,3 +17,109 @@ func VaultConfigMiddleware(conf *sonic.VaultConfig) gin.HandlerFunc {
 		c.Next()
 	}
 }
+
+// OpendaxConfigMiddleware middleware to set kaigara config to gin context
+func OpendaxConfigMiddleware(config *sonic.OpendaxConfig) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		c.Set("OpendaxConfig", config)
+		c.Next()
+	}
+}
+
+// GlobalVaultServiceMiddleware middleware to set global vault service to gin context
+func GlobalVaultServiceMiddleware(vaultService *vault.Service) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		c.Set("GlobalVaultService", vaultService)
+		c.Next()
+	}
+}
+
+// AuthMiddleware middleware to verify bearer token
+func AuthMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		// Get bearer token from header
+		authHeader := strings.Split(c.GetHeader("Authorization"), "Bearer ")
+		if len(authHeader) != 2 {
+			c.Abort()
+			c.JSON(http.StatusBadRequest, gin.H{"error": "Authorization header not found"})
+			return
+		}
+
+		// Get global vault service
+		vaultService, err := GetGlobalVaultService(c)
+		if err != nil {
+			c.Abort()
+			c.JSON(http.StatusNotFound, gin.H{"error": err.Error()})
+			return
+		}
+
+		key := "sonic_public_key"
```

Comment:
There is no reason to put this logic in the AuthMiddleware.
It doesn't have anything to do with the Auth!
It only relates to the registration endpoint, so it should in the handler directly.
Also we said we need to retrieve the 3 public keys: peatio, barong and sonic.

Reply:
Finally we will inject those keys into the environment, so we can skip the vault part here and directly read them from env variables, let's say named:
- PEATIO_PUBLIC_KEY
- BARONG_PUBLIC_KEY
- SONIC_PUBLIC_KEY

Reply:
Ok, I can get all public keys from ENV directly. But

1. Which public key used to verify JWT in Sonic’s admin endpoint? Do I need another ENV name JWT_PUBLIC_KEY for JWT verification only?
2. How I know which public keys to be used during platform registration (peatio, barong or sonic)?

Reply:
Fixed.